gfootball/env/observation_processor.py

```diff
@@ -178,22 +239,44 @@ def write_dump(name, trace, config):
         writer.write('FRAME: %d' % frame_cnt)
         writer.write('TIME: %f' % (o._time - time))
         sticky_actions = football_action_set.get_sticky_actions(config)
-        sticky_actions_field = 'left_agent_sticky_actions'
-        if len(o[sticky_actions_field]) == 0:
-          sticky_actions_field = 'right_agent_sticky_actions'
-        assert len(sticky_actions) == len(o[sticky_actions_field][0])
-        active_direction = None
-        for i in range(len(sticky_actions)):
-          if sticky_actions[i]._directional:
-            if o[sticky_actions_field][0][i]:
-              active_direction = sticky_actions[i]
-          else:
-            writer.write('%s: %d' % (sticky_actions[i]._name,
-                                     o[sticky_actions_field][0][i]))
-        writer.write('DIRECTION: %s' % ('NONE' if active_direction is None
-                                        else active_direction._name))
-        if 'action' in o._trace['debug']:
-          writer.write('ACTION: %s' % (o['action'][0]._name))
+
+        players_actions = {}
+        for team in ['left', 'right']:
+          sticky_actions_field = '%s_agent_sticky_actions' % team
+          players_actions[team] = {}
+          for player in range(len(o[sticky_actions_field])):
+            assert len(sticky_actions) == len(o[sticky_actions_field][player])
+            player_idx = o['%s_agent_controlled_player' % team][player]
+            players_actions[team][player_idx] = {'team': team,
+                                                 'player_idx': str(player_idx)}
+            active_direction = None
+            for i in range(len(sticky_actions)):
+              if sticky_actions[i]._directional:
+                if o[sticky_actions_field][player][i]:
+                  active_direction = sticky_actions[i]
+              else:
+                players_actions[team][player_idx][sticky_actions[i]._name] = \
+                    o[sticky_actions_field][player][i]
+
+            # Info about direction
+            players_actions[team][player_idx]['DIRECTION'] = \
+                '-' if active_direction is None else active_direction._name
+            if 'action' in o._trace['debug']:
+              # Info about action
+              players_actions[team][player_idx]['ACTION'] = \
+                  o['action'][player]._name
+
+        no_players = len(players_actions['left']) + \
+                     len(players_actions['right'])
+        if no_players > 1:
+            pprint_players_actions(writer, players_actions)
```

Comment:
Why not to use the same format for both single and multi player modes? I don't see any good reason to complicate the code. It is fine to change the format of dumps.